src/lib/services/generation/ChapterBatchService.ts

```
@@ -0,0 +1,184 @@
+
+    for (let i = 0; i < boundaries.length; i++) {
+      if (callbacks.isCancelled()) {
+        log('Batch cancelled during execution', {
+          chaptersCreated: chapters.length,
+          planned: totalChapters,
+        })
+        return { chapters, cancelled: true }
+      }
+
+      const boundary = boundaries[i]
+      
```

ESLint

[error] Delete `······`

(prettier/prettier)